### PR TITLE
Add support for docker_registry and kms_key_name

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,18 @@ steps:
 - `build_environment_variables_file`: (Optional) Path to a local YAML file
   containing variables. See 'env_vars_file' for syntax.
 
+- `docker_repository`: (Optional) User managed repository created in Artifact
```
Comment (bharathkkb) [inline README.md:105]: Lets clarify expected input is of form `projects/${PROJECT}/locations/${LOCATION}/repositories/${REPOSITORY}`

Comment (bharathkkb) [reply README.md:105]: Also looks like cross project image repos as are not allowed. Maybe we can just link to https://cloud.google.com/sdk/gcloud/reference/beta/functions/deploy#--docker-repository

**Review by bharathkkb — COMMENTED** [review]
LGTM, minor suggestion

File: action.yaml [hunk 2]
```diff
@@ -147,6 +147,19 @@ inputs:
...
+    description: |-
+      User managed repository created in Artifact Registry.
+    required: false
+
+  kms_key_name:
+    description: |-
+      Resource name of a Google Cloud KMS crypto key used to encrypt/decrypt
+      function resources. If specified, you must also provide an artifact
+      registry repository using the 'docker_repository' field that was created
+      with the same key.
```
Comment (bharathkkb) [inline action.yaml:160]: Can we validate this at input time and fail fast rather than after API call?

File: README.md (hunk 1 above)
Comment (sethvargo) [reply README.md:105]: Done

File: action.yaml (hunk 2 above)
Comment (sethvargo) [reply action.yaml:160]: Done

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**